grammars/css.cson

```diff
@@ -20,294 +20,1218 @@
+  {
+    'include': '#combinators'
+  }
+  {
     'include': '#selector'
   }
   {
-    'begin': '(@)charset\\b'
-    'captures':
+    'include': '#at-rules'
+  }
+  {
+    'include': '#rule-list'
+  }
+  {
+    'match': ';'
+    'name': 'punctuation.terminator.rule.css'
+  }
+  {
+    'match': '}'
+    'name': 'punctuation.section.end.css'
+  }
+]
+'repository':
+  'at-rules':
+    'patterns': [
+      {
+        # @charset
+        'begin': '\\A(?:\\xEF\\xBB\\xBF)?(?i:(?=\\s*@charset\\b))'
+        'end': ';|(?=$)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.terminator.rule.css'
+        'name': 'meta.at-rule.charset.css'
+        'patterns': [
+          {
+            'captures':
+              '1':
+                'name': 'invalid.illegal.not-lowercase.charset.css'
+              '2':
+                'name': 'invalid.illegal.leading-whitespace.charset.css'
+              '3':
+                'name': 'invalid.illegal.no-whitespace.charset.css'
+              '4':
+                'name': 'invalid.illegal.whitespace.charset.css'
+              '5':
+                'name': 'invalid.illegal.unquoted-identifier.charset.css'
+              '6':
+                'name': 'invalid.illegal.unclosed-string.charset.css'
+              '7':
+                'name': 'invalid.illegal.unexpected-characters.charset.css'
+            'match': '''(?x)        # Possible errors:
+              \\G
+              ((?!@charset)@\\w+)   # Not lowercase (@charset is case-sensitive)
+              |
+              \\G(\\s+)             # Preceding whitespace
+              |
+              (@charset\\S[^;]*)    # No whitespace after @charset
+              |
+              (?<=@charset)         # Before quoted charset name
+              (\\x20{2,}|\\t+)      # More than one space used, or a tab
+              |
+              (?<=@charset\\x20)    # Beginning of charset name
+              ([^"';]+)             # Not quoted
+              |
+              (["'][^"']+$)         # Unclosed quote
+              |
+              (?<=["'])             # After charset name
+              ([^;]+)               # Unexpected junk instead of semicolon
+            '''
+          }
+          {
+            'captures':
+              '1':
+                'name': 'keyword.control.at-rule.charset.css'
+              '2':
+                'name': 'punctuation.definition.keyword.css'
+            'match': '((@)charset)(?=\\s)'
+          }
+          {
+            'include': '#string-double'
+          }
+          {
+            'include': '#string-single'
+          }
+          {
+            'include': '#comment-block'
+          }
+        ]
+      }
+      {
+        # @import
+        'begin': '(?i)((@)import)(?:\\s+|$|(?=[\'"]|/\\*))'
+        'beginCaptures':
+          '1':
+            'name': 'keyword.control.at-rule.import.css'
+          '2':
+            'name': 'punctuation.definition.keyword.css'
+        'end': '\\s*(?=;|})'
```

Comment:
Because people these days have some inexplicable phobia of semicolons and refuse to touch them.

(No reason, will amend)